calicoctl/tests/st/calicoctl/test_convert.py

```diff
@@ -25,53 +26,70 @@
 logger = logging.getLogger(__name__)
 
 convert_files = [
-        ("test-data/v1/bgppeer-global.yaml",),
-        ("test-data/v1/bgppeer-node.yaml",),
-        ("test-data/v1/bgppeer-node2.yaml",),
-        ("test-data/v1/multi-resource.yaml",),
-        ("test-data/v1/node.yaml",),
-        ("test-data/v1/test3.yaml",),
-        ("test-data/v1/migration/bgppeer.yaml",),
-        ("test-data/v1/migration/hostendpoint.yaml",),
-        ("test-data/v1/migration/ippool.yaml",),
-        ("test-data/v1/migration/node.yaml",),
-        ("test-data/v1/migration/policy.yaml",),
-        ("test-data/v1/migration/profile.yaml",),
-        ("test-data/v1/migration/workloadendpoint.yaml",),
-        ("test-data/k8s/k8s-networkpolicy.yaml",),
+        ("test-data/convert/input/v1/bgppeer-global.yaml",),
+        ("test-data/convert/input/v1/bgppeer-node.yaml",),
+        ("test-data/convert/input/v1/bgppeer-node2.yaml",),
+        ("test-data/convert/input/v1/multi-resource.yaml",),
+        ("test-data/convert/input/v1/node.yaml",),
+        ("test-data/convert/input/v1/test3.yaml",),
+        ("test-data/convert/input/v1/migration/bgppeer.yaml",),
+        ("test-data/convert/input/v1/migration/hostendpoint.yaml",),
+        ("test-data/convert/input/v1/migration/ippool.yaml",),
+        ("test-data/convert/input/v1/migration/node.yaml",),
+        ("test-data/convert/input/v1/migration/policy.yaml",),
+        ("test-data/convert/input/v1/migration/profile.yaml",),
+        ("test-data/convert/input/v1/migration/workloadendpoint.yaml",),
+        ("test-data/convert/input/k8s/k8s-networkpolicy.yaml",),
+        ("test-data/convert/input/k8s/k8s-networkpolicy-invalid.yaml",),
+        ("test-data/convert/input/k8s/k8s-networkpolicy-multiple.yaml",),
+        ("test-data/convert/input/k8s/k8s-networkpolicy-multiple-invalid.yaml",),
     ]
 
 class TestCalicoctlConvert(TestBase):
     """
     Test calicoctl convert
     """
 
-    def _test_convert_successful(self, filename, format="yaml"):
+    def _test_convert(self, filename, format="yaml"):
         """
         Test convert successfully
         """
         # Convert the file
         rc = calicoctl("convert -o %s -f %s" % (format, filename))
-        rc.assert_no_error()
 
-        # With the converted data to a temp file
-        with open("/tmp/converted", 'w') as f:
-            f.write(rc.output)
+        # Get expected conversion output filename (file in output/ dir with
+        # yaml or json extension)
+        output_filename = (filename.replace("input", "output").split(".")[0]
+                           + "." + format)
```

Comment:
is there a reason to split out the file format here? I would have thought that `input` wouldn't be a part of any of the file format so you wouldn't need to split it out.

Reply:
I'm doing 2 things: replacing the 'input' dir with 'output' and then replacing the extension with format, e.g.:
'test-data/convert/input/v1/bgppeer-global.yaml' -> 'test-data/convert/output/v1/bgppeer-global.json'

Reply:
Oh my bad, I misread it and thought that the convert command above already converted the filename. This should be fine.